pkg/protocol/gossip/warpsync.go

```diff
@@ -324,3 +324,46 @@ func (w *WarpSyncMilestoneRequester) Cleanup() {
 
 	w.traversed = make(map[string]struct{})
 }
+
+// RequestMilestoneRange requests up to N milestones nearest to the current confirmed milestone index.
+// Returns the number of milestones requested.
+func (w *WarpSyncMilestoneRequester) RequestMilestoneRange(rangeToRequest int, onExistingMilestoneInRange func(milestone *storage.CachedMilestone), from ...milestone.Index) int {
+	var requested int
+
+	// make sure we only request what we don't have
+	startingPoint := w.syncManager.ConfirmedMilestoneIndex()
+	if len(from) > 0 {
```

Comment:
compare from[0] to startingPoint?

Reply:
I removed the confusing comment, but the logic will stay as it is for now.